ONSdigital/census-field-service
# Cr 569 - census-field-service

# Motivation and Context

This change is required in order for the Field Service to call library functions, in the census-int-case-api-client, which access the RM case service api. 
These functions will return the case details, and questionnaire id, for a given case id.


# What has changed

The following class has been created that calls the library functions:
uk.gov.ons.ctp.integration.censusfieldsvc.service.impl.LauncherServiceImpl

The following interface has been created for the class to implement:
uk.gov.ons.ctp.integration.censusfieldsvc.service.LauncherService




# How to test?

Test the changes by doing as follows:
Run rabbit using docker from census-rh-service (docker-compose up -d)
Run the FakeCaseServiceApplication.java within Eclipse (NB. You will need to use the version that's currently in the CR-569 branch as this contains the fake endpoint for the questionnaire id)
Run this service (mvn spring-boot:run)
Go to the following URL http://localhost:8177/c45de4dc-3c3b-11e9-b210-d663bd873d13 (NB. If all is well this will just be a blank page)
Check the logging output to see if the case details are as follows:
CaseContainerDTO(caseRef=123123, arid=2344266233, estabArid=AABBCC, estabType=ET, uprn=1347459999, createdDateTime=Sun Apr 14 13:45:26 BST 2019, addressLine1=Napier House, addressLine2=11 Park Street, addressLine3=Parkhead, townName=Glasgow, postcode=G1 2AA, organisationName=ON, addressLevel=E, abpCode=AACC, latitude=41.40338, longitude=2.17403, oa=EE22, lsoa=x1, msoa=x2, lad=H1, state=null, caseEvents=[], id=c45de4dc-3c3b-11e9-b210-d663bd873d13, caseType=HH, region=E)
And also check the logging to see if the questionnaire id is as follows:
QuestionnaireIdDTO(questionnaireId=1110000010)

NB. The logging might not show any case details or questionnaire id as  the log.with(casedetails) and the log.with(questionnaireId) methods put these objects into JSON format, so you may need to 
add more logging statements to the getEqUrl method in LauncherServiceImpl e.g.
log.info("The case details received: " + caseDetails);
log.info("The questionnaire id received: " + questionnaireId);
